core/src/test/java/org/apache/calcite/test/UdfTest.java

```diff
@@ -423,7 +447,8 @@ private CalciteAssert.AssertThat withUdf() {
 
   @Test void testUdfOverloadedNullable() {
     final CalciteAssert.AssertThat with = withUdf();
-    with.query("values (\"adhoc\".count_args(),\n"
+    with.with(SqlConformanceEnum.MYSQL_5)
```

Comment:
why is this change needed?

Reply:
When we use the default conformance, its allowNiladicParentheses method returns false, and functions without parameters should not use parentheses. In order to ensure that the existing test cases can run normally, we need to explicitly set the conformance and ensure that the allowNiladicParentheses of the conformance returns true.

Reply:
perhaps a comment explaining that MYSQL_5 allows nilladic functions without parentheses will help people understand the point. That is not obvious.

Reply:
Thank you @mihaibudiu, I will add comment for this change.